sympy/ntheory/factor_.py

```diff
@@ -1591,6 +1592,13 @@ def eval(cls, n):
             for p, k in factors.items():
                 t *= (p - 1) * p**(k - 1)
             return t
+        elif (n.is_Symbol or isinstance(n,Expr)):
```

Comment:
this can all be condensed (I believe) to

```
elif not isinstance(n, Expr) or n.is_integer is False or n.is_positive is False:
    raise error
```